### PR TITLE
Add azkaban flowId tag to metrics

### PR DESCRIPTION
**Pull request by pcadabam**

### CODE REVIEW

File: gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java [hunk 1]
```diff
@@ -96,6 +99,13 @@ public JobContext(Properties jobProps, Logger logger) throws Exception {
         previousJobState.getPropAsInt(ConfigurationKeys.JOB_FAILURES_KEY, 0));
 
     if (GobblinMetrics.isEnabled(jobProps)) {
+      Configuration configuration = new Configuration();
+      String flowId = configuration.get("azkaban.flow.flowid");
```
Comment (liyinan926) [inline gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java]: The key may not exist, so you should check first if it exists.

File: gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java [hunk 2]
```diff
@@ -96,6 +99,13 @@ public JobContext(Properties jobProps, Logger logger) throws Exception {
...
+
+      if (StringUtils.isNotBlank(flowId)) {
```
Comment (liyinan926) [inline gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java]: Please use Guava's `Strings.isNullOrEmpty` instead.

File: gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java (hunk 1 above)
Comment (liyinan926) [inline gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java]: Please define a String constant for `azkaban.flow.flowid` in `ConfigurationKeys`.

Comment (liyinan926) [inline gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java]: Actually I think it's better to put it into a class say called `TagNames` under `gobblin-core/src/main/java/gobblin/metrics/`.

File: gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java (hunk 2 above)
Comment (pcadabam) [inline gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java]: Strings.isNullOrEmpty does not check white spaces.
it returns false for " ", I want it to be true.

File: gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java (hunk 1 above)
Comment (pcadabam) [inline gobblin-runtime/src/main/java/gobblin/runtime/JobContext.java]: 1. Defined a constant for azkaban.flow.flowid
2. Moved tag names to gobblin-core/src/main/java/gobblin/metrics/TagNames.java

**Comment by liyinan926** [comment]
LGTM.

**Pull request merged**